plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseHiveConnectorTest.java

```diff
@@ -2102,6 +2102,39 @@ private void testBucketedTable(Session session, HiveStorageFormat storageFormat,
         assertFalse(getQueryRunner().tableExists(session, tableName));
     }
 
+    @Test
+    public void testBucketedTableWithBoundedVarchar()
+    {
+        String tableName = "test_bucketed_table_with_bounded_varchar_" + randomTableSuffix();
+        assertUpdate(
+                """
+                CREATE TABLE %s (bucket_key varchar(20), other_data varchar(20))
+                WITH (
+                    bucketed_by = ARRAY[ 'bucket_key' ],
+                    bucket_count = 10)
+                """.formatted(tableName));
+        assertUpdate(
+                """
+                INSERT INTO %s
+                SELECT clerk as bucketed_by, orderpriority as other_data
+                FROM tpch.sf1.orders
```

Comment:
Do we need sf1 ? Can we do the test with tiny scale instead ?
sf1 can be slow in CI